File: src/test/ruby/NativeTypesAction.rb
require 'java'
require 'TrifaceJRubyAction.rb'

# dummy action to set some basic values for testing.
# sets basic ruby types
class NativeTypesAction < TrifaceJRubyAction
  
	def execute()
		@objectmap["string"] = "hello world!"
		@objectmap["boolean"] = true
		@objectmap["int"] = 1
		@objectmap["float"] = 1.0
		@objectmap["list"] = [1,1,2,3,5]
		@objectmap["map"] = {"key1" => "val1", "key2" => 2}
		return getMap()
	end
	
end

File: src/test/js/NativeTypesAction.js
// map -- this.objectMap
action.execute=function(){
	this.objectmap.put("string","hello world!");
	this.objectmap.put("boolean", true);
	this.objectmap.put("int",1);
	this.objectmap.put("float",1.0);
	this.objectmap.put("list", [1,1,2,3,5]);
	
	tmpMap = new HashMap();
	tmpMap.put("key1","val1");
	tmpMap.put("key2",2)
	
	this.objectmap.put("map", tmpMap)
	return this.getMapInternal();
};
